### ISSUE TITLE
Feature suggestion: Allow to change how auto-clock behaves in "step-by-step" mode

### ISSUE THREAD

**Issue by rinart73** [logisim-evolution/logisim-evolution#1603]
A way to change signal propagation speed and gate delays would be nice. It could help with observing things step-by-step.
Or alternatively allow to automatically advance clock every X steps. Right now the "single step propagation" option seems to advance auto ticking clocks at random (depends on the input?), which it seems creates inconsistent behaviour between "real time" and "step by step" modes. Like for example getting oscillation in "step by step" but never getting it in real time.

Or am I misunderstanding this and the clock ticks only after inputs were fully processed which allows to roughly track how much time did the full update take?

**Comment by maehne** [comment]
Indeed, single-step propagation allows you to see how events (i.e., signal changes) get processed by blocks and propagate through the system. Only once signals stabilise, time advances in form of clock ticks. That's how digital simulators work. Logisim-evolution's simulation engine has a very simplified notion of time. The smallest time step is always a clock step. That is why it is not feasible to model shorter gate propagation delays. It would require a lot of work to change this central piece of Logisim-evolution, for which we currently don't have any resources. The original author of Logisim, Carl Burch, had also recognised these shortcomings and decided to therefore reimplement a successor to Logisim from scratch. Unfortunately, this effort stalled before it could replace Logisim in terms of features/functionality.

I will therefore mark this issue as won't fix. If you disagree, feel free to reopen the issue.
